apps/api/lib/utils/isValidBase64.ts

```diff
@@ -0,0 +1,17 @@
+export function isValidBase64(input: string): boolean {
+  try {
+    let base64Data = input;
+    // Check if the input starts with a data URI scheme (data:content/type;base64,...)
+    if (input.startsWith("data:")) {
+      const parts = input.split(",");
+      if (parts.length === 2) {
+        // metadata and base64Data
+        base64Data = parts[1]; // Extract the base64 data from the second part
+      }
+    }
+    const buffer = Buffer.from(base64Data, "base64").toString("base64");
+    return buffer === base64Data;
+  } catch (error) {
+    return false;
+  }
```

Comment:
couldn't we replace this whole thing with a regexp ? 

Also I think the naming of the function is unclear, we are here checking if the data inside of the data-uri string is encoded in base64.

Reply:
Sure we can. but why do yo prefer using regex over a native lib?
--
I tried to name the extension method to be as generic as possible to accommodate all base64-encoding-string checks, not just for images.